Gems/Atom/RPI/Code/Include/Atom/RPI.Public/Pass/RenderPass.h

```diff
@@ -142,6 +142,8 @@ namespace AZ
             PipelineStatisticsResult m_statisticsResult;
 
             // The device index the pass should run on. Can be invalid if it doesn't matter.
+            // Please note that the m_deviceIndex might not be the actual running device index if it is invalid
+            // because the actually running device index is specified in RenderPass::FrameBeginInternal if m_deviceIndex is invalid.
             int m_deviceIndex{RHI::MultiDevice::InvalidDeviceIndex};
```

Comment:
This should be moved to Pass.h/removed, or is there a reason to keep another device index in `RenderPass`?

Reply:
I am considering another PR to remove the `m_deviceIndex` here and allowing setting the device index in the `ScopeProducer`, but yes I can also do it in this PR